sdk/core/core-client/src/urlHelpers.ts

```diff
@@ -103,7 +103,15 @@ function appendPath(url: string, pathToAppend?: string): string {
     pathToAppend = pathToAppend.substring(1);
   }
 
-  newPath = newPath + pathToAppend;
+  const searchStart = pathToAppend.indexOf("?");
+  if (searchStart !== -1) {
+    const path = pathToAppend.substr(0, searchStart);
+    const search = pathToAppend.substr(searchStart + 1);
+    newPath = newPath + path;
+    parsedUrl.search = parsedUrl.search ? parsedUrl.search + "&" + search : search;
```

Comment:
```suggestion
    parsedUrl.search = parsedUrl.search ? `${parsedUrl.search}&${search}` : search;
```

Reply:
I wonder if there are any cases where this wouldn't concatenate correctly? It's probably not worth parsing `search` with URLSearchParams and using that to set each key/value pair on `parsedUrl.searchParams`...

Reply:
Yes I too had thought about parsing and appending using the helpers you already added but chose not to do so. `appendQueryParams()` will be called after `appendPath()` so the quirk case will be handled there.